src/Compilers/Core/Portable/InternalUtilities/WeakReferenceExtensions.cs

```diff
@@ -1,19 +1,23 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
 using System;
+using System.Diagnostics.CodeAnalysis;
 
 namespace Roslyn.Utilities
 {
     // Helpers that are missing from Dev11 implementation:
     internal static class WeakReferenceExtensions
     {
-        public static T GetTarget<T>(this WeakReference<T> reference) where T : class
+        [return: MaybeNull]
```

Comment:
`T?` is only allowed when constrained to `struct` or `class`, not for `class?`.

Reply:
@RikkiGibson What's the reason for that?

Reply:
Oh I guess the answer is "it's already null, so it's meaningless", but then I guess that means my real question is "if T is known to be null, why the MaybeNull"?

Reply:
The compiler does not allow the types used in this class to be represented. The most accurate representation here would be `T?`. `[MaybeNull] T` is the same thing except with flow analysis disabled.

Reply:
> if T is known to be null

T is not known to be nullable. `string` and `string?` are both valid substitutions for `class?`.

Reply:
Ah, right so my answer to my own question was wrong. Question to @RikkiGibson stands then. :-)

Reply:
So, this is because this is what LDM decided: https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-08-20.md#class-constraint

That being said, I personally think that `T?` should be allowed. @gafter or @agocke, do you remember more in depth arguments from that LDM as to why we don't allow `T?` when `T : class?`